src/modules/rewards/components/ActivityVerificationButton.tsx

```diff
@@ -33,6 +38,7 @@ export const ActivityVerificationButton = ({
   setErrorMessage,
   userId,
   isLoadingActivity,
+  startingLabel,
```

Comment:
Either all the label should be handled inside or passed from outside. 

Half logic for labels in inside this component and rest is decided from outside. This is incorrect and very confusing.